ProcessLib/Parameter/CurveScaledParameter.h

```diff
@@ -29,13 +29,24 @@ struct CurveScaledParameter final : public Parameter<T> {
     }
 
     bool isTimeDependent() const override { return true; }
+
+    void setParameter(ParameterBase const* parameter)
+    {
+        if (dynamic_cast<Parameter<double> const*>(parameter) == nullptr)
+        {
+            OGS_FATAL(
+                "Parameters other then for double type are not supported.");
```

Comment:
But the CurveScaledParameter is parametrized by `T`.

Reply:
Fixing this part, but the actual problem is in the create method, where there is only double type allowed: https://github.com/ufz/ogs/blob/master/ProcessLib/Parameter/CurveScaledParameter.cpp#L38-L40

Same for all other types.